File: chat-uikit/src/main/java/io/agora/chat/uikit/widget/chatrow/EaseChatRowUnknown.java
package io.agora.chat.uikit.widget.chatrow;

import android.content.Context;
import android.text.Spannable;
import android.text.Spanned;
import android.text.style.URLSpan;
import android.view.View;
import android.widget.TextView;
import android.widget.TextView.BufferType;

import io.agora.chat.ChatMessage;
import io.agora.chat.TextMessageBody;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.manager.EaseDingMessageHelper;
import io.agora.chat.uikit.utils.EaseSmileUtils;


public class EaseChatRowUnknown extends EaseChatRow {
	private TextView contentView;

    public EaseChatRowUnknown(Context context, boolean isSender) {
		super(context, isSender);
	}

    public EaseChatRowUnknown(Context context, ChatMessage message, int position, Object adapter) {
		super(context, message, position, adapter);
	}

	@Override
	protected void onInflateView() {
		inflater.inflate(!showSenderType ? R.layout.ease_row_received_message
                : R.layout.ease_row_sent_message, this);
	}

	@Override
	protected void onFindViewById() {
		contentView = (TextView) findViewById(R.id.tv_chatcontent);
	}

    @Override
    public void onSetUpView() {
        contentView.setText(getContext().getString(R.string.ease_chat_unknown_type));
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/widget/chatrow/EaseChatRowUnknown.kt
package com.hyphenate.easeui.widget.chatrow

import android.content.Context
import android.util.AttributeSet
import android.widget.TextView
import com.hyphenate.easeui.R

class EaseChatRowUnknown @JvmOverloads constructor(
    private val context: Context,
    private val attrs: AttributeSet? = null,
    private val defStyleAttr: Int = 0,
    isSender: Boolean = false
) : EaseChatRow(context, attrs, defStyleAttr, isSender) {
    private val contentView: TextView? by lazy { findViewById(R.id.tv_chatcontent) }

    override fun onInflateView() {
        inflater.inflate(
            if (!isSender) R.layout.ease_row_received_message else R.layout.ease_row_sent_message,
            this
        )
    }

    override fun onSetUpView() {
        contentView?.text = context.getString(R.string.ease_chat_unknown_type)
    }
}